catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/action/SourceActionProviderImpl.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.impl.action;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.source.Source;
+import java.net.MalformedURLException;
+import java.net.URL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Convert a JSON string to a map of source identifiers to action URLs. */
+public class SourceActionProviderImpl implements ActionProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SourceActionProviderImpl.class);
+
+  private String actionProviderId;
+
+  private String title;
+
+  private String url;
+
+  private String description;
+
+  private String sourceId;
+
+  public SourceActionProviderImpl(String actionProviderId) {
+    this.actionProviderId = actionProviderId;
+  }
+
+  @Override
+  public synchronized <T> Action getAction(T subject) {
+
+    if (!(subject instanceof Source)) {
+      return null;
+    }
+
+    Source source = (Source) subject;
+
+    if (source.getId().equals(sourceId)) {
+      return toAction();
+    }
+
+    return null;
+  }
+
+  @Override
+  public String getId() {
+    return actionProviderId;
+  }
+
+  public void setSourceId(String sourceId) {
+    this.sourceId = sourceId;
+  }
+
+  public void setTitle(String title) {
+    this.title = title;
+  }
+
+  public void setUrl(String url) {
+    this.url = url;
+  }
+
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  private Action toAction() {
+    try {
+      return new ActionImpl(getId(), title, description != null ? description : "", new URL(url));
```

Comment:
Seems like the description setter could handle the null-checking logic so you don't have to do it here.

Reply:
moved the logic to the setter